### PR TITLE
MTDSA-4357 Create controller, route and ITs

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: app/v1/controllers/requestParsers/validators/TriggerBSASValidator.scala [hunk 1]
```diff
@@ -38,47 +38,38 @@ class TriggerBSASValidator @Inject()(val currentDateProvider: CurrentDateProvide
...
-    List(
-      DateValidation.validate(req.startDate, StartDateFormatError),
-      DateValidation.validate(req.endDate, EndDateFormatError)
-    )
-  }
-
-  private def taxYearValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
-    val req = data.body.json.as[BSAS]
+    val startDate: JsLookupResult = (data.body.json \ "accountingPeriod" \ "startDate")
+    val endDate: JsLookupResult = (data.body.json \ "accountingPeriod" \ "endDate")
```
Comment (MarkAKelly) [inline app/v1/controllers/requestParsers/validators/TriggerBSASValidator.scala]: These parantheses are unnecessary here.

File: test/v1/fixtures/TriggerBsasRequestBodyFixtures.scala [hunk 2]
```diff
@@ -85,4 +87,168 @@ object TriggerBsasRequestBodyFixtures {
...
+  val response = Json.parse("""{
+      |"id" : "c75f40a6-a3df-4429-a697-471eeec46435"
+      |}""".stripMargin)
+
+  val hateoasResponseForSE = (nino: String) => s"""
+      |{
+      |  "id": "c75f40a6-a3df-4429-a697-471eeec46435",
+      |  "links":[
+      |    {
+      |      "href":"/individuals/self-assessment/accounting-summary/${nino}/self-employment/c75f40a6-a3df-4429-a697-471eeec46435",
```
Comment (MarkAKelly) [inline test/v1/fixtures/TriggerBsasRequestBodyFixtures.scala]: Unecessary parantheses again :slightly_smiling_face:

File: test/v1/fixtures/TriggerBsasRequestBodyFixtures.scala [hunk 3]
```diff
@@ -85,4 +87,168 @@ object TriggerBsasRequestBodyFixtures {
...
+      |  ]
+      |}
+    """.stripMargin
+
+  val hateoasResponseForProperty = (nino: String) => s"""
+      |{
+      |  "id": "c75f40a6-a3df-4429-a697-471eeec46435",
+      |  "links":[
+      |    {
+      |      "href":"/individuals/self-assessment/accounting-summary/${nino}/property/c75f40a6-a3df-4429-a697-471eeec46435",
```
Comment (MarkAKelly) [inline test/v1/fixtures/TriggerBsasRequestBodyFixtures.scala]: and here

File: test/v1/controllers/requestParsers/validators/TriggerBSASValidatorSpec.scala [hunk 4]
```diff
@@ -30,46 +30,48 @@ class TriggerBSASValidatorSpec extends UnitSpec {
...
     MockCurrentDateProvider.getCurrentDate().returns(date)
   }
 
-
   "running validation" should {
     "return no errors" when {
 
       "a valid self employment is supplied" in new SetUp {
 
+        println(scala.Console.YELLOW + triggerBsasRawDataBody() + scala.Console.RESET)
```
Comment (MarkAKelly) [inline test/v1/controllers/requestParsers/validators/TriggerBSASValidatorSpec.scala]: :scream:

Comment (james-work-account) [reply test/v1/controllers/requestParsers/validators/TriggerBSASValidatorSpec.scala]: You saw nothing...

**Pull request merged**